src/Middleware/HttpLogging/src/HttpLoggingMiddleware.cs

```diff
@@ -45,13 +47,22 @@ public HttpLoggingMiddleware(RequestDelegate next, IOptionsMonitor<HttpLoggingOp
                 throw new ArgumentNullException(nameof(options));
             }
 
-            if (logger == null)
+            if (loggerFactory == null)
             {
-                throw new ArgumentNullException(nameof(logger));
+                throw new ArgumentNullException(nameof(loggerFactory));
             }
 
+            _filterOptions = filterOptions;
             _options = options;
-            _logger = logger;
+
+            // By default, disable sending events to W3CLogger
+            // TODO - it seems odd to add this rule so late. Would it be better to add it in the extension method that adds HttpLoggingMiddleware?
+            // Somewhere else?
+            _filterOptions.CurrentValue.Rules.Add(new LoggerFilterRule(null, "Microsoft.AspNetCore.W3CLogging", LogLevel.None, null));
```

Comment:
Here's the rule I mentioned above, which excludes W3CLogging by default. Also uses the arbitrary string `Microsoft.AspNetCore.W3CLogging` again.